### ISSUE TITLE
Unable to use DbConfigurationTypeAttribute if DBConfiguration is loaded into the appdomain

### ISSUE THREAD

**Issue by amccool** [aspnet/EntityFramework6#263]
I have a simple case where I wish to setup a DBContext to use a particular DbConfiguration using the attribute on the DbContext rather than via the app.config

However even though there are no other usages of EF in a project I continue to get

`System.InvalidOperationException: The default DbConfiguration instance was used by the Entity Framework before the 'KeyValueDbConfiguration' type was discovered. An instance of 'KeyValueDbConfiguration' must be set at application start before using any Entity Framework features or must be registered in the application's config file`

If I add 
`  <entityFramework codeConfigurationType="xx.KeyValueDbConfiguration, xxx">`

to the app.config everything is fine.

Is there a way to find what loaded the default DbConfiguration, and stop it?

Here is the repo I am working with: https://github.com/amccool/Orleans.StorageProviders.SimpleSQLServerStorage/tree/140-with-sessionid

**Comment by phatcher** [comment]
@amccool I'll think you find that DbConfiguration behaviour is global e.g. I had a project recently with two DbContexts using separate DbConfiguration types, when I attempt to create the context, the second one would complain about the other DbConfiguration.

**Comment by amccool** [comment]
going from memory here.   I believe I figured this out.    I was not using the 'correct' constructor for the dbcontext.
`public KeyValueDbContext(DbConnection dbConnection, bool contextOwnsConnection)
            : base(dbConnection, contextOwnsConnection) { }`

once I switched to this constructor things where loading 'better'.   As it were, the documentation is good, but like WCF, its tough to correlate the error message to the documentation for the wanted behavior.

Lets treat this issue and a question - and close it.
BTW - having the source available was really the saving grace.  👍 for Microsoft open sourcing
